src/renderer/components/ft-community-post/ft-community-post.js

```diff
@@ -0,0 +1,394 @@
+      return this.$store.getters.getHideVideoViews
+    },
+
+    addWatchedStyle: function () {
+      return this.watched && !this.inHistory
+    },
+
+    favoritesPlaylist: function () {
+      return this.$store.getters.getFavorites
+    },
+
+    favoriteIconTheme: function () {
+      return this.inFavoritesPlaylist ? 'base favorite' : 'base'
+    }
+  },
+  mounted: function () {
+    this.parseVideoData()
+  },
+  methods: {
+    toggleSave: function () {
+      if (this.inFavoritesPlaylist) {
+        this.removeFromPlaylist()
+      } else {
+        this.addToPlaylist()
+      }
+    },
+
+    handleOptionsClick: function (option) {
+      switch (option) {
+        case 'history':
+          if (this.watched) {
+            this.removeFromWatched()
+          } else {
+            this.markAsWatched()
+          }
+          break
+        case 'copyYoutube':
+          navigator.clipboard.writeText(this.youtubeShareUrl)
+          showToast(this.$t('Share.YouTube URL copied to clipboard'))
+          break
+        case 'openYoutube':
+          if (this.usingElectron) {
+            const shell = require('electron').shell
+            shell.openExternal(this.youtubeUrl)
+          }
+          break
+        case 'copyYoutubeEmbed':
+          navigator.clipboard.writeText(this.youtubeEmbedUrl)
+          showToast(this.$t('Share.YouTube Embed URL copied to clipboard'))
+          break
+        case 'openYoutubeEmbed':
+          if (this.usingElectron) {
+            const shell = require('electron').shell
+            shell.openExternal(this.youtubeEmbedUrl)
+          }
+          break
+        case 'copyInvidious':
+          navigator.clipboard.writeText(this.invidiousUrl)
+          showToast(this.$t('Share.Invidious URL copied to clipboard'))
+          break
+        case 'openInvidious':
+          if (this.usingElectron) {
+            const shell = require('electron').shell
+            shell.openExternal(this.invidiousUrl)
+          }
+          break
+        case 'copyYoutubeChannel':
+          navigator.clipboard.writeText(this.youtubeChannelUrl)
+          showToast(this.$t('Share.YouTube Channel URL copied to clipboard'))
+          break
+        case 'openYoutubeChannel':
+          if (this.usingElectron) {
+            const shell = require('electron').shell
+            shell.openExternal(this.youtubeChannelUrl)
+          }
+          break
+        case 'copyInvidiousChannel':
+          navigator.clipboard.writeText(this.invidiousChannelUrl)
+          showToast(this.$t('Share.Invidious Channel URL copied to clipboard'))
+          break
+        case 'openInvidiousChannel':
+          if (this.usingElectron) {
+            const shell = require('electron').shell
+            shell.openExternal(this.invidiousChannelUrl)
+          }
+          break
+      }
+    },
+
+    // For Invidious data, as duration is sent in seconds
+    calculateVideoDuration: function (lengthSeconds) {
```

Comment:
If #2794 is merged before this PR please use the `formatDurationAsTimestamp` instead.